core/cloudflow-build-support/src/main/scala/com/lightbend/cloudflow/buildtool/Scaffold.scala

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.cloudflow.buildtool
+
+import java.nio.file._
+import java.io._
+import com.typesafe.config.{ Config, ConfigFactory }
+import cloudflow.blueprint.deployment._
+import cloudflow.blueprint.deployment.ApplicationDescriptorJsonFormat._
+import spray.json._
+
+import scala.util._
+
+object Scaffold {
+
+  def getDescriptorsOrFail[T](descriptors: Iterable[(T, Try[RuntimeDescriptor])])(
+      error: String => Unit): Iterable[(T, RuntimeDescriptor)] = {
```

Comment:
The function has no semantics of failure as part of its type signature. Should we name the function `getDescriptorsOrFail` or change it simply to `getDescriptors` ? It throws but usually we don't add that as a semantics in the name of the function. One option would be to change the signature to `Try[Iterable[..` and wrap the exception in a `Try` ..

Reply:
Those functions have been extracted all "as-is" from the sbt plugin.
This is not fresh new code being added to the code-base.

Reply:
also is good to throw the exception to the user in those cases, not much we can do with it.
By handling it gracefully we are going just to mask the original error